### ISSUE TITLE
raise might eat errors when used with strformat

### ISSUE THREAD

**Issue by alaviss** [nim-works/nimskull#112]
### Example

```nim
import std/strformat

proc exec(cmd: string, args: varargs[string]) =
  let x = @[cmd] & args
  raise newException(CatchableError):
    fmt"{x}"
  # Only referring to a "broken" variable would this issue happen
```

### Actual Output

```
test.nim(5, 3) Error: only a 'ref object' can be raised
```

### Expected Output

The real cause of the error (invalid expression for `x`) should show up instead

```
test.nim(4, 18) Error: type mismatch: got <seq[string], varargs[string]>
but expected one of:
proc `&`[T](x, y: seq[T]): seq[T]
  first type mismatch at position: 2
  required type for y: seq[T]
  but expression 'args' is of type: varargs[string]
proc `&`[T](x: T; y: seq[T]): seq[T]
  first type mismatch at position: 2
  required type for y: seq[T]
  but expression 'args' is of type: varargs[string]
proc `&`[T](x: seq[T]; y: T): seq[T]
  first type mismatch at position: 2
  required type for y: T
  but expression 'args' is of type: varargs[string]
5 other mismatching symbols have been suppressed; compile with --showAllMismatches:on to see them

expression: @[cmd] & args
```

### Additional information

* Mainline does not have this issue

**Comment by beef331** [comment]
Error report order issue with `--errorMax:0` the error is there just not presented first
```
/home/jason/nimskull/bin/abuse.nim(5, 3) Error: only a 'ref object' can be raised
/home/jason/nimskull/bin/abuse.nim(4, 18) Error: type mismatch: got <seq[string], varargs[string]>
but expected one of:
proc `&`[T](x, y: seq[T]): seq[T]
  first type mismatch at position: 2
  required type for y: seq[T]
  but expression 'args' is of type: varargs[string]
proc `&`[T](x: T; y: seq[T]): seq[T]
  first type mismatch at position: 2
  required type for y: seq[T]
  but expression 'args' is of type: varargs[string]
proc `&`[T](x: seq[T]; y: T): seq[T]
  first type mismatch at position: 2
  required type for y: T
  but expression 'args' is of type: varargs[string]
5 other mismatching symbols have been suppressed; compile with --showAllMismatches:on to see them
```

**Comment by zerbina** [comment]
Fixed by #1261, errors are now properly propagated for `raise` statements.
